lib/tensorflow/tensor.rb

```diff
@@ -182,17 +182,39 @@ def ruby_array_to_c(array, type)
   #   - Value of the element contained in the specified position in the tensor.
   #
   def getval(dimension)
-    raise("Invalid dimension array passed as input.",ShapeError) if dimension.length != dimensions.length
+    raise("Invalid dimension array passed as input.",ShapeError) if dimension.length != shape.length
     (0..dimension.length-1).each do |i|
-      raise("Invalid dimension array passed as input.",ShapeError) if dimension[i] > dimensions[i] || dimension[i] < 1 || !(dimension[i].is_a? Integer)
+      raise("Invalid dimension array passed as input.",ShapeError) if dimension[i] > shape[i] || dimension[i] < 1 || !(dimension[i].is_a? Integer)
     end
-    sum = dimension[dimension.length - 1]  - 1
-    prod = dimensions[dimensions.length - 1]
+    sum = dimension.last - 1
+    prod = shape.last
     (0..dimension.length - 2).each do |i|
        sum += (dimension[dimension.length - 2 - i] - 1) * prod
-       prod *= dimensions[dimensions.length - 2 - i]
+       prod *= shape[shape.length - 2 - i]
     end
 
     flatten[sum]
   end
+
+  private
+
+  #
+  # Recursively finds the shape of the input array.
+  #
+  # * *Returns* :
+  #   - Dimension array `[[2], [4]].shape` => `[2, 1]`
+  #
+  # TODO: Handle non-rectangular arrays and raise error if mixed data types
+  def self.set_shape(value)
```

Comment:
@zverok My idea is to follow Ruby naming convention. I meant that the layout and design of the classes, how data is passed around, etc. should as a starting point be done similar to the Python implementation in order for the user to expect a similar behavior using both Swig wrappers and for us not to reinvent the wheel. As far as I can see, Python's `set_shape` actually does analyze the incoming shape it just happens through the [`merge_with`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/framework/tensor_shape.py#L551) method. I think we'll be best off getting a very small subset of the API to work first than implementing all of the functionality of Tensorflow in one step 😅